config/localTemplate.go

```diff
@@ -506,6 +506,10 @@ type Local struct {
 	// and APIs related to broadcasting transactions, and enables APIs which can retrieve detailed information
 	// from ledger caches and can control the ledger round.
 	EnableFollowMode bool `version[27]:"false"`
+
+	// EnableTransactionSimulator enables the /v2/transaction/simulate endpoint. This functionality
+	// is disabled by default.
+	EnableTransactionSimulator bool `version[27]:"false"`
```

Comment:
This adds the new config option and defaults it to false. Some alternates which I decided against were:
* Associating the simulation endpoint with the existing `EnableDeveloperAPI`. Decided against because simulating transactions is useful beyond just development, and most node operators have this disabled (from personal experience)
* Adding is new option, but defaulting it to true. This is what I would prefer, but since we definitely don't want relays to enable this, it's a bit safer to default to false for everyone

Reply:
this feels like a PM decision

Reply:
Agree on PM, but want to point out two things that could help make that decision.

1. Even if relays mistakenly activate it, the normal token based API access prevents randos from using it, I think.
2. It doesn't have to be a bool. Consider a tri-state like "yes", "no", "if-not-relay", where "if-not-relay" is the default. (better names would be great).

Reply:
We could also introduce tunable rate-limiting mechanisms.

Reply:
I'm not that familiar with the defaults for our other API endpoints (so same question as JJ). Without knowing that I lean towards defaulting to off. The consequence to it being a surprise that it's off is that then you have to turn it on. The consequence to it being a surprise that it's on is that your node is getting unexpected requests/load and you have to figure out what's going on and maybe it even crashes, and that feels worse. 

Also, hopefully the introduction of the config profiles will make this kind of decision feel less important, since then we can disable it for the relay profile, enable it for the follower profile...

Reply:
I am not in favor of the tri-state because of the added complexity, and I think we should use profiles for that (but correct me if I'm thinking about it wrong). 

The tunable rate-limiting mechanism sounds complex, and also sounds like something that API providers already handle themselves (I think?) so we don't need to do it for them.

Reply:
https://github.com/algorand/sandbox/pull/171 linking this issue to the resolution of this thread

Reply:
If I'm reading this thread correctly, we are still in favor of **adding the new `EnableTransactionSimulator` flag and defaulting it to false**?

For boxes, we do not consider it a `EnableDeveloperAPI`, but we have a [config param](https://github.com/algorand/go-algorand/pull/5159/files#diff-0ea0d4e61032f93218f6e8b6662c4047cf99a56f11c07d9a004d54dc81c61f3eR487) to limit the maximum number of boxes that can be retrieved by algod (defaults to 100K).

Reply:
I pushed a new commit [3c4fe45](https://github.com/algorand/go-algorand/pull/5159/commits/3c4fe452997b0b27a1421c772feb5ca9b744f906) that defaults this config parameter to true, except on relay nodes that are specified in `NetAddress`. I think there is some precedent for having a new simulation config - even if we explicitly don't guard slow APIs in relays today, we do (for instance) offer a config to limit how many boxes can be retrieved at a time. A nice bonus is that we can also turn it off for relay nodes and default it to on for non-relay nodes.